mzLib/FlashLFQ/ResultsReading/MzLibExtensions.cs

```diff
@@ -53,12 +51,51 @@ public static List<Identification> MakeIdentifications(this IQuantifiableResultF
                         proteinGroups.Add(allProteinGroups[info.proteinAccessions]);
                     }
                 }
-                Identification id = new Identification(file, baseSequence, modifiedSequence, monoisotopicMass, ms2RetentionTimeInMinutes, precursurChargeState, proteinGroups);
+                Identification id = new Identification(spectraFile, baseSequence, modifiedSequence, monoisotopicMass, ms2RetentionTimeInMinutes, precursurChargeState, proteinGroups);
                 identifications.Add(id);
-
             }
 
             return identifications;
         }
+
+        private static Dictionary<string, SpectraFileInfo> makeSpectraFileDict(this IQuantifiableResultFile quantifiable, List<SpectraFileInfo> spectraFiles)
+        {
+            Dictionary<string, SpectraFileInfo> allSpectraFiles = new Dictionary<string, SpectraFileInfo>();
+
+            // 1. from list of SFIs create a list of strings that contains each full file path w/ extension
+            List<string> fullFilePaths = new List<string>();
+            foreach (SpectraFileInfo spectraFileInfo in spectraFiles)
+            {
+                fullFilePaths.Add(spectraFileInfo.FullFilePathWithExtension);
+            }
+
+            // 2. call quantifiableresultfile.filenametofilepath and get stringstring dict
+            Dictionary<string, string> allFiles = quantifiable.FileNameToFilePath(fullFilePaths);
+
+            // 3. using stringstring dict create a string spectrafileinfo dict where key is same b/w dicts and value fullfilepath is replaced spectrafileinfo obj
+            SpectraFileInfo spectraFile = null;
+            foreach (var file in allFiles)
+            {
+                string key = file.Key;
+                string filePath = file.Value;
+                // FirstOrDefault matches the 1st elt from spectraFiles w/ specified filePath
+                SpectraFileInfo? matchingSpectraFile = spectraFiles.FirstOrDefault(spectraFileInfo => spectraFileInfo.FullFilePathWithExtension == filePath);
+
+                if (matchingSpectraFile != null)
+                {
+                    if (allSpectraFiles.TryGetValue(key, out var existingSpectraFile))
```

Comment:
This can be simplified. Just check if allSpectraFiles contains the key. If not, add the key/value. If it already contains it, don't do anything. Can be combined into one if statement